### PR TITLE
Update submodule tips

### LINKED ISSUE #1214: Automatically update submodules when other repos push to `main`
### Plone Documentation versions affected

- [ ] 5.x
- [X] 6.x

### Description

Currently whenever there is a push to `master` for either repo `plone/volto` or `plone/plone.restapi`, we manually need to remember to update the submodule. See https://6.docs.plone.org/contributing/index.html#editing-external-package-documentation, Step 7.

```
cd ../..

# For plone.restapi
git add submodules/plone.restapi
git commit -m "Update submodules/plone.restapi tip"

# For volto
git add submodules/volto
git commit -m "Update submodules/volto tip"

git push
```

It would be great to handle this automatically in GitHub Actions.

### PR DESCRIPTION
**Pull request by esteele**
Add an action to update the submodule tips that can be called by volto/restapi/api.

Closes #1214

### CODE REVIEW

File: .github/workflows/update_submodule.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      # Pull latest plone.restapi
+      - name: Pull latest plone.restapi
+        run: |
+          cd submodules/plone.restapi
+          git pull
+
+      # Pull latest Volto
+      - name: Pull latest Volto
+        run: |
+          cd submodules/volto
```
Comment (stevepiercy) [inline .github/workflows/update_submodule.yml:30]: I am not sure whether `cd` resets to the root of `documentation` for each command. I guess we will see.

File: .github/workflows/update_submodule.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          git pull
+
+      # Update, commit, and push
+      - name : Update tips
+        run: |
+          git status
+          git add submodules/plone.api submodules/plone.restapi submodules/volto
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git diff --exit-code && git commit -m "Update submodules tip" && git push
```
Comment (stevepiercy) [inline .github/workflows/update_submodule.yml:40]: I assume that this workflow only fires when there are diffs in the `docs/src/` directories of each submodule. For example, in Volto, Netlify has this configuration.

https://github.com/plone/volto/blob/5e6fab072500364dd239d44fa6bec76e627bb749/netlify.toml#L4-L5

**Review by stevepiercy — DISMISSED** [review]

File: .github/workflows/update_submodule.yml (hunk 1 above)
Comment (esteele) [reply .github/workflows/update_submodule.yml:30]: It does.

**Review by stevepiercy — APPROVED** [review]

**Comment by stevepiercy** [comment]
@esteele I just updated my local docs repo and its submodules, but when I do a `git status`:

```console
$ git status
On branch 6-dev
Your branch is up to date with 'origin/6-dev'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   submodules/plone.api (new commits)
        modified:   submodules/plone.restapi (new commits)
        modified:   submodules/volto (new commits)

no changes added to commit (use "git add" and/or "git commit -a")
```

I expected that this PR would also commit the submodules to `6-dev`. The [latest failure was from 2 days ago](https://github.com/plone/documentation/actions/runs/3255922707/jobs/5345744536), but there have been [more recent merges to Volto](https://github.com/plone/volto/actions). It looks like the submodule repos need to be fired on pushes to their `master` branches, for example see [volto](https://github.com/plone/volto/actions/runs/3255862511/workflow).

**Comment by esteele** [comment]
Ok. I’ll look into it. On Oct 17, 2022, at 8:30 PM, Steve Piercy ***@***.***> wrote:﻿
@esteele I just updated my local docs repo and its submodules, but when I do a git status:
$ git status
On branch 6-dev
Your branch is up to date with 'origin/6-dev'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   submodules/plone.api (new commits)
        modified:   submodules/plone.restapi (new commits)
        modified:   submodules/volto (new commits)

no changes added to commit (use "git add" and/or "git commit -a")
I expected that this PR would also commit the submodules to 6-dev. The latest failure was from 2 days ago, but there have been more recent merges to Volto. It looks like the submodule repos need to be fired on pushes to their master branches, for example see volto.

**Pull request merged**